src/components/step-sequence/step-sequence-item/step-sequence-item.js

```diff
@@ -0,0 +1,35 @@
+import Icon from 'components/icon';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import './step-sequence-item.scss';
+
+const baseClass = 'carbon-step-sequence-item';
+const classes = state => `${baseClass} ${baseClass}--${state}`;
+
+const stepMarker = (state, indicator) => {
+  return state === 'complete' ? <Icon type='tick' /> : indicator;
+};
+
+const StepSequenceItem = ({ label, state, indicator }) => (
+  <li className={ classes(state) }>
+    <span className='carbon-step-sequence-item__label'>
+      <span className='carbon-step-sequence-item__indicator'>{ stepMarker(state, indicator) }</span>
+      { label }
+    </span>
+  </li>
+);
+
+StepSequenceItem.propTypes = {
+  label: PropTypes.string,
```

Comment:
Instead of `label` can you just used `children`? Just in case they ever want to use something more complicated than a string it makes it a bit more intuitive.